packages/atomic/src/components/atomic-search-box-v1/atomic-search-box.tsx

```diff
@@ -0,0 +1,124 @@
+import SearchIcon from 'coveo-styleguide/resources/icons/svg/search.svg';
+import ClearIcon from 'coveo-styleguide/resources/icons/svg/clear.svg';
+import {Component, h, State} from '@stencil/core';
+import {SearchBox, SearchBoxState, buildSearchBox} from '@coveo/headless';
+import {
+  Bindings,
+  BindStateToController,
+  InitializeBindings,
+} from '../../utils/initialization-utils';
+import {Button} from '../common/button';
+
+/**
+ * The `atomic-search-box` component creates a search box with built-in support for suggestions.
+ */
+@Component({
+  tag: 'atomic-search-box-v1', // TODO: remove v1
+  styleUrl: 'atomic-search-box.pcss',
+  shadow: true,
+})
+export class AtomicSearchBox {
+  @InitializeBindings() public bindings!: Bindings;
+
+  private searchBox!: SearchBox;
+
+  @BindStateToController('searchBox')
+  @State()
+  private searchBoxState!: SearchBoxState;
+  @State() public error!: Error;
+  @State() hasFocus = false;
+
+  public initialize() {
+    this.searchBox = buildSearchBox(this.bindings.engine, {
+      options: {
+        numberOfSuggestions: 8, // TODO: handle when adding query suggestion component
```

Comment:
This will be populated with an event from a child element, I presume ?

Reply:
Yes as designed in the document, with various children